src/views/Experiences/ExperiencesComponents/ExperiencesComponent.vue

```diff
@@ -42,15 +49,121 @@ const filteredExperiences = computed(() => {
 const handleAddExperience = () => {
   emit('toggleAddExperience');
 };
+
+// Function to check if the vote exists
+const checkExistingVote = (exid: number) => {
+  return votes.details.some(vote => vote.experience_id === exid);
+};
+
+// Function to find the vote ID with a given experience ID
+const findVoteId = (exid: number) => {
+  return votes.details.find(vote => vote.experience_id === exid);
+}
+
+// Checks vote data for pretty button functions
+const isUpvote = (exid: number) => {
+  const vote = findVoteId(exid);
+  return vote && vote.helpfulness === 'up'; // Ensure it exists and is an upvote
+};
+
+const isDownvote = (exid: number) => {
+  const vote = findVoteId(exid);
+  return vote && vote.helpfulness === 'down'; // Ensure it exists and is a downvote
+};
+
+// Function to fetch all votes for a given experience ID
+const retrieveVote = async (exid: number) => {
+  if (auth.currentUser) {
+    try {
+      const response = await fetch(`${apiUrl}/experiences/${exid}/votes`);
+      if (!response.ok) {
+      throw new Error(`Response status: ${response.status}`);
+      }
+      votes.details = await response.json();
+      console.log(votes.details);
+    } catch (error) {
+      console.error(error);
+    }
+  } else {
+    console.log("User not logged in.")
+  }
+}
+
+// Handles the addition, editing, and deletion of votes
+const handleVote = async (exid:number, vote:string) => {
+  // If user is logged in and the vote does not exist, add vote
+  if (auth.currentUser && !checkExistingVote(exid) && vote) {
+    try {
+      await fetch(`${apiUrl}/experiences/${exid}/votes`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        //vote data
+        userID: auth.currentUser.uid,
+        experienceID: exid,
+        vote: vote,
+      }),
```

Comment:
Not sure yet, but if memory serves me right, we might not need to send the userID because it goes with the cookie and gets authenticated on the backend. It should then be accessible on the backend.